### PR TITLE
Add a simple_tag for views to render a list of values

### PR DESCRIPTION
**Pull request by MyPyDavid**
This tag should return a readable string from a list of `Values` in a `View` template. 
The default keyword arguments for this simple tag are separator `sep = ','` and the second to last separator `sep_last = ', and'`.
The name of the tag is currently `render_custom_inline_list`.

For example for a list of possible values (like in a optionset)  `A, B, C, D` it should return:
> A, B, C, and D
> A, B, and C
> A and B
> A

In the View for example:
```html
<p> Im Projekt entstehen Daten
{% for dataset in datasets %}
    {% if forloop.last and not forloop.first %}und {% endif %}
    im Bereich <i>{{ dataset.value }}</i> durch
    {% get_set_values dataset 'project/dataset/creation_methods' as texts %}
    {% render_custom_inline_list values=texts sep=';' sep_last='und' %}
{% endfor %}
</p>
```

Co-authored-by: @jwindeck

## Description

Related issue: #ISSUE_NUMBER

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Checklist

- [x] I have read the [contributor guide](https://github.com/rdmorganiser/rdmo/blob/main/CONTRIBUTING.md).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by Zack-83 — APPROVED** [review]

**Comment by Zack-83** [comment]
addresses https://github.com/rdmorganiser/rdmo-catalog/issues/245

File: rdmo/views/templatetags/view_tags.py [hunk 1]
```diff
@@ -111,6 +111,33 @@ def get_set(context, attribute, set_prefix='', project=None):
...
+
+    if len(values) == 1:
+        return values[0]['value_and_unit']
+
+    if len(values) == 2:
+        if sep_last.startswith(', '):  # for english sep_last
+            sep_last = sep_last.split(', ')[-1]
+        return f"{values[0]['value_and_unit']} {sep_last} {values[1]['value_and_unit']}"
+
+    _text = ''
```
Comment (jochenklar) [inline rdmo/views/templatetags/view_tags.py:129]: Why the underscore? Looks odd.

File: rdmo/views/templatetags/view_tags.py [hunk 2]
```diff
@@ -111,6 +111,33 @@ def get_set(context, attribute, set_prefix='', project=None):
     return get_sets(context, attribute, set_prefix=set_prefix, project=project)
 
 
+@register.simple_tag
+def render_custom_inline_list(values=None, sep=',', sep_last=', and'):
```
Comment (jochenklar) [inline rdmo/views/templatetags/view_tags.py:115]: I would call the tag differently, all the `render_` tags have more or less the same signature. I also dislike custom (filler word). Maybe `create_inline_list` or `join_list` / `join_values`?

Comment (jochenklar) [reply rdmo/views/templatetags/view_tags.py:115]: Hmm I like `join_values` more I think.

File: rdmo/views/templatetags/view_tags.py [hunk 3]
```diff
@@ -111,6 +111,33 @@ def get_set(context, attribute, set_prefix='', project=None):
...
+    values = values or []
+
+    if not values:
+        return ''
+
+    if len(values) == 1:
+        return values[0]['value_and_unit']
+
+    if len(values) == 2:
+        if sep_last.startswith(', '):  # for english sep_last
```
Comment (jochenklar) [inline rdmo/views/templatetags/view_tags.py:125]: This is ugly, this should be done with another argument.

File: rdmo/views/templatetags/view_tags.py (hunk 2 above)
Comment (jochenklar) [reply rdmo/views/templatetags/view_tags.py:115]: Also `seperator` is better than `sep`.

**Review by jochenklar — CHANGES_REQUESTED** [review]

File: rdmo/views/templatetags/view_tags.py (hunk 2 above)
Comment (Zack-83) [reply rdmo/views/templatetags/view_tags.py:115]:
```suggestion
def join_values(values=None, separator=',', sep_last=', and'):
```

File: rdmo/views/templatetags/view_tags.py [hunk 4]
```diff
@@ -111,6 +111,33 @@ def get_set(context, attribute, set_prefix='', project=None):
...
+        if n == len(values) - 1 and _text:
+            _text += f"{sep_last} "
+        _text += value['value_and_unit']
+        if n < len(values) - 2:
+            _text += f"{sep} "
+        else:
+            _text += ' '
+    return _text
+
+
```
Comment (Zack-83) [inline rdmo/views/templatetags/view_tags.py:140]:
```suggestion
            sep_last = separator_last.split(', ')[-1]
        return f"{values[0]['value_and_unit']} {separator_last} {values[1]['value_and_unit']}"

    text = ''
    for n,value in enumerate(values):
        if n == len(values) - 1 and text:
            text += f"{separator_last} "
        _text += value['value_and_unit']
        if n < len(values) - 2:
            text += f"{separator} "
        else:
            text += ' '
    return text


```

Comment (Zack-83) [inline rdmo/views/templatetags/view_tags.py:140]: Ignoranter Versuch, nicht getestet: können wir vielleicht in der `if`-Bedingung nur den Separator ersetzen?

```suggestion
    if len(values) == 2:
        separator_last = separator_last.replace(', ',' ')  # for english sep_last

    text = ''
    for n,value in enumerate(values):
        if n == len(values) - 1 and text:
            text += f"{separator_last} "
        _text += value['value_and_unit']
        if n < len(values) - 2:
            text += f"{separator} "
        else:
            text += ' '
return text


```

Comment (Zack-83) [inline rdmo/views/templatetags/view_tags.py:140]: Ignoranter Versuch, nicht getestet.
* Können wir vielleicht in der `if`-Bedingung nur den Separator ersetzen?
* Ich vermisse in der `for`-Schleife den Fall `len(values) - 2`: war es ein Tippfehler?
* Ist der Fall `len(values) == 1` nicht automatisch bedeckt, wenn `n == len(values) - 1` (also `else`)?

```suggestion
    if len(values) == 2:
        separator_last = separator_last.replace(', ',' ') # for english sep_last

    text = ''
    for n,value in enumerate(values):
        if n == len(values) - 2 and text:
            text += f"{separator_last} "
        text += value['value_and_unit']
        if n < len(values) - 2:
            text += f"{separator} "
        else:
            text += ''
    return text


```

File: rdmo/views/templatetags/view_tags.py (hunk 2 above)
Comment (MyPyDavid) [reply rdmo/views/templatetags/view_tags.py:115]: yes, naming still needed to be discussed.
* `create_inline_list_with_separator`
* `join_values_with_separator`
* `join_values_inline_with_separator`
?

Comment (MyPyDavid) [reply rdmo/views/templatetags/view_tags.py:115]: -> `join_values_inline`

Comment (jochenklar) [reply rdmo/views/templatetags/view_tags.py:115]: `join_values_inline(values=None, seperator=',', seperator_two='and', seperator_last=', and')`

**Review by jochenklar — APPROVED** [review]

**Comment by MyPyDavid** [comment]
thanks, looks much better with the `separator_two`

**Pull request merged**